Client/U2FHelper.swift

```diff
@@ -0,0 +1,615 @@
+        }
+    }
+
+    private func signInner(authenticated: Bool, keyHandle: String, challengeParam: NSData, appParam: NSData, result: (String -> ()), error: ((U2FErrorCode, String) -> ())) {
+        guard authenticated else {
+            error(U2FErrorCode.OTHER_ERROR, "User presence test failed")
+            return
+        }
+
+        // Check the appParam is known
+        guard self.knownKey(keyHandle, forAppParam: appParam) else {
+            error(U2FErrorCode.OTHER_ERROR, "Unkown appID for key handle")
+            return
+        }
+
+        // Unwrap the private key from the key handle
+        guard let handle = NSData(base64URLEncodedString: keyHandle, options: NSDataBase64DecodingOptions()) else {
+            error(U2FErrorCode.OTHER_ERROR, "Unable to decode key handle")
+            return
+        }
+        let keyLen = handle.length - (kParamLength + kIVLength + kMACLength)
+        let iv = handle.subdataWithRange(NSMakeRange(kParamLength, kIVLength))
+        let mac = handle.subdataWithRange(NSMakeRange(kParamLength + kIVLength, kMACLength))
+        let encryptedPrivate = handle.subdataWithRange(NSMakeRange(kParamLength + kIVLength + kMACLength, keyLen))
+
+        guard let keys = self.getKeys() else {
+            error(U2FErrorCode.OTHER_ERROR, "Unable to process key handle")
+            return
+        }
+
+        guard keys.verify(hmac: mac, ciphertextB64: encryptedPrivate) else {
+            error(U2FErrorCode.OTHER_ERROR, "Corrupted key handle")
+            return
+        }
+
+        guard let privB64 = keys.decrypt(encryptedPrivate, iv: iv) else {
+            error(U2FErrorCode.OTHER_ERROR, "Unable to unwrap key handle")
+            return
+        }
+
+        guard let privBytes = NSData(base64URLEncodedString: privB64, options: NSDataBase64DecodingOptions()) else {
+            error(U2FErrorCode.OTHER_ERROR, "Unable to decode key handle")
+            return
+        }
+
+        guard let privateKey = ECDSAPrivateKey(binaryRepresentation: privBytes, group: .P256) else {
+            error(U2FErrorCode.OTHER_ERROR, "Unable to parse key handle")
+            return
+        }
+
+        // Compute the signature
+        // message = appParam || 0x01 || counter(4) || challengeParam
+        let counter = self.getNextCounter()
+        let presenceAndCounter = NSMutableData(bytes: [UInt8](arrayLiteral: 1), length: 1)
+        presenceAndCounter.appendData(counter)
+
+        let message = NSMutableData()
+        message.appendData(appParam)
+        message.appendData(presenceAndCounter)
+        message.appendData(challengeParam)
+        let sig = privateKey.signMessage(message)
+
+        // Assemble the response data
+        let responseData = presenceAndCounter
+        responseData.appendData(sig)
+        result(responseData.base64URLEncodedStringWithOptions(NSDataBase64EncodingOptions()))
+    }
```

Comment:
Similar to the above, should we zeroize `privB64` and `privBytes` on our way out?